globus_action_provider_tools/flask/helpers.py

```diff
@@ -211,7 +211,16 @@ def pydantic_input_validation(
     try:
         validator(**action_input)
     except ValidationError as ve:
-        messages = [f"Field '{'.'.join(e['loc'])}': {e['msg']}" for e in ve.errors()]
+        messages = []
+        for error in ve.errors():
+            path = []
+            for location in error["loc"]:
+                if isinstance(location, str):
+                    path.append(f".{location}")
+                else:
+                    path.append(f"[{location}]")
+            field = "".join(path)[1:]  # Remove the leading period.
```

Comment:
https://github.com/globusonline/globus-flows/blob/main/src/globus_flows/api/routes/blueprint/pydantic_errors.py#L31-L44

![image](https://github.com/globus/action-provider-tools/assets/113056046/b3551e93-6d0d-4139-9c14-754659c7c9c0)

Reply:
Technically we do it slightly differently in flows I guess.

* APs: `foo[0].bar`
* Flows: `$.foo[0].bar`

Reply:
I _knew_ I'd seen this brilliant formatting code somewhere before! :heart: